django/contrib/admin/templatetags/admin_list.py

```diff
@@ -201,18 +203,18 @@ def link_in_col(is_first, field_name, cl):
         except ObjectDoesNotExist:
             result_repr = empty_value_display
         else:
-            empty_value_display = getattr(attr, 'empty_value_display', empty_value_display)
+            empty_value_display = mark_safe(getattr(attr, 'empty_value_display', empty_value_display))
             if f is None or f.auto_created:
                 if field_name == 'action_checkbox':
                     row_classes = ['action-checkbox']
                 allow_tags = getattr(attr, 'allow_tags', False)
                 boolean = getattr(attr, 'boolean', False)
-                if boolean or not value:
-                    allow_tags = True
                 result_repr = display_for_value(value, empty_value_display, boolean)
-                # Strip HTML tags in the resulting text, except if the
-                # function has an "allow_tags" attribute set to True.
                 if allow_tags:
+                    warnings.warn(
+                        "The allow_tags attribute is deprecated. Return a SafeString instance by "
+                        "using format_html, format_html_join or mark_safe instead.",
+                        RemovedInDjango20Warning, stacklevel=2)
```

Comment:
I don't know that stacklevel=2 will be of any use here (could you check?). Instead it would probably be helpful to include the attr name in the message (some comment with the warning above).